comfyui-truss/README.md

```diff
@@ -0,0 +1,316 @@
+
+Inside ComfyUI, you can save workflows as a JSON file. However, the regular JSON format that ComfyUI uses will not work. Instead, the workflow has to be saved in the API format. Here is how you can do that:
+
+First, go to ComfyUI and click on the gear icon for the project
+
+![gear_icon](images/comfyui-screenshot-1.png)
+
+Next, checkmark the box which says `Enable Dev Mode Options`
+
+![enable_dev_mode_options](images/comfyui-screenshot-2.png)
+
+Now, if you go back to the project you will see a new option called `Save(API Format)`. This is the one you want to use to save your workflow. Using this method you can save any ComfyUI workflow as a JSON file in the API format.
+
+![save_api_format](images/comfyui-screenshot-3.png)
+
+
+## Setting up the project
+
+First, clone this repository:
+
+```sh
+git clone https://github.com/basetenlabs/truss-examples/
+cd comfyui-truss
+```
+
+For your ComfyUI workflow, you probably used one or more models. Those models need to be defined inside truss. From the root of the truss project, create and open the file inside the data directory called `data/model.json`. This file will contain all of the models that need to get downloaded in order for your ComfyUI workflow to run. Here is an example of the contents of this file:
+
+```json
+[
+    {
+        "url": "https://huggingface.co/stabilityai/stable-diffusion-xl-base-1.0/resolve/main/sd_xl_base_1.0.safetensors",
+        "path": "checkpoints/sd_xl_base_1.0.safetensors"
+    },
+    {
+        "url": "https://huggingface.co/diffusers/controlnet-canny-sdxl-1.0/resolve/main/diffusion_pytorch_model.fp16.safetensors",
+        "path": "controlnet/diffusers_xl_canny_full.safetensors"
+    }
+]
+```
+
+In this case, I have 2 models: SDXL and a controlnet. Each model needs to have 2 things, `url` and `path`. The `url` is the location for downloading the model. The `path` is where this model will get stored inside the Truss. For the path, follow the same guidelines as used in ComfyUI. Models should get stored inside `checkpoints`, controlnets should be stored inside `controlnet`, etc.
+
+
+## Deployment
+
+Before deployment:
+
+1. Make sure you have a [Baseten account](https://app.baseten.co/signup) and [API key](https://app.baseten.co/settings/account/api_keys).
+2. Install the latest version of Truss: `pip install --upgrade truss`
+
+With `comfyui-truss` as your working directory, you can deploy the model with:
+
+```sh
+truss push
+```
+
+Paste your Baseten API key if prompted.
+
+For more information, see [Truss documentation](https://truss.baseten.co).
+
+## Model Inference
+
+The main thing we need for inference is the JSON workflow we exported in step 1. Inside the JSON workflow file, there might be some inputs such as the positive prompt or negative prompt that are hard coded. We want these inputs to be dynamically sent to the model, so we can use handlebars to templatize them. Here is an example of a JSON workflow with templatized inputs:
+
+```json
```

Comment:
I wonder if we should move the workflow to be `data/` in the truss as well instead of input. What do you think?